### PR TITLE
Feature: App User Context | Spring Testing | Notification Automation | Continuous Integration | Frontend Integration

### PR DESCRIPTION
**Pull request by skushagra**
This PR makes the following changes in the codebase

1. Adds app user context, it creates a user everytime a new FCM token is recieved from the frontend and registers the user in the app.
2. Adds Tests for controller and service. They are not yet ready for CI but can be run locally.
3. Adds automatic notification support for appeals and books.
4. Adds any required API for the frontend.

### CODE REVIEW

File: .github/workflows/gradle.yml
Comment (AttAditya) [inline .github/workflows/gradle.yml:1]: I don't seem to be able to identify any changes in this file...

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -44,4 +44,4 @@ jobs:
     #     gradle-version: '8.5'
     #
     # - name: Build with Gradle 8.5
-    #   run: gradle build
+    #   run: gradle build
```
Comment (AttAditya) [inline .github/workflows/gradle.yml:47]: I am not able to understand what has changed here...

File: .gitignore [hunk 2]
```diff
@@ -36,6 +36,5 @@ out/
 ### VS Code ###
 .vscode/
 
-### Creds ###
-.token
-.creds
```
Comment (AttAditya) [inline .gitignore:41]: Why have we removed creds?

Comment (skushagra) [reply .gitignore:41]: No such file is present in our codebase.

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (skushagra) [reply .github/workflows/gradle.yml:47]: This is for CI. We have not changed anything here.

**Review by 2k4sm — APPROVED** [review]
lgtm

**Pull request merged**